controllers/front/fail.php

```diff
@@ -124,8 +124,19 @@ public function initContent()
             ]);
         }
 
-        $this->setTemplate(
-            sprintf('module:%s/views/templates/front/order_fail.tpl', $this->module->name)
+        $this->warning[] = $this->l('We couldn\'t authorize your payment. Please try again.');
+        $this->redirectWithNotifications(
```

Comment:
be aware this function does not exist on P.S 1.6 version - I suggest adding a function with same name to your parent controller which has identical logic as this one